sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/dofn/ReadChangeStreamPartitionDoFn.java

```diff
@@ -130,8 +130,7 @@ public TimestampRange initialRestriction(@Element PartitionMetadata partition) {
     final com.google.cloud.Timestamp startTimestamp = partition.getStartTimestamp();
     // Range represents closed-open interval
     final com.google.cloud.Timestamp endTimestamp =
-        ofTimeSecondsAndNanos(
-            partition.getEndTimestamp().getSeconds(), partition.getEndTimestamp().getNanos() + 1);
+        TimestampUtils.next(partition.getEndTimestamp());
```

Comment:
Makes sure there is no overflow here by using `TimestampUtils.next` instead of simply adding `1ns`